src/node/rpc/frontend.h

```diff
@@ -284,7 +286,18 @@ namespace ccf
             return ctx->serialise_response();
           }
 
-          switch (tx.commit())
+          std::function<kv::Version()> version_resolver = nullptr;
+          if (prescribed_commit_version != kv::NoVersion)
+          {
+            CCF_ASSERT(
+              consensus->type() == ConsensusType::BFT, "Wrong consensus type");
+            version_resolver = [&]() {
+              tables.next_version();
+              return prescribed_commit_version;
+            };
+          }
+
+          switch (tx.commit(version_resolver, max_conflict_version))
```

Comment:
Can you rewrite this as:

```
kv::CommitResult result;
if (prescribed_commit_version != kv::NoVersion)
{
  version_resolver = ...;
  result = tx.commit(version_resolver, max_conflict_version);
}
else
{
  result = tx.commit();
}

switch (result)
```

I think it helps readability to make it clear there's really a simpler `tx.commit()` path here, without working backwards to compare the default values and confirm that's what's being called.